### PR TITLE
Modifications when batch_size of data and label are not the same

### PR DESCRIPTION
**Pull request by jonbakerfish**
Modifications for the python training code, when the of the batch_size of input data and label are not the same. Related to issue [#214](https://github.com/dmlc/mxnet/issues/214). Modifications for fast-rcnn training.

For now we can call `model.fit(X=your_data_iter)` to train the network where `your_data_iter` has different batch_size for the image and label.

### CODE REVIEW

**Comment by piiswrong** [comment]
Can you work around this by reshaping the label into (data_batch_size, label_batch_size/data_batch_size, ...)

**Comment by jonbakerfish** [comment]
@piiswrong Can I add a flatten layer for the label and used the flattened label as input for the loss layer?

**Comment by pluskid** [comment]
Yes you can add a flat layer to flat labels (see the current symbol constructed for char-LSTM for example). There is however one caveat for that: once you start applying operators on the label, during prediction time, if you want to bind an executor without providing label shape (because there is no label), it will fail because the shape inference for the label pipeline will fail. Of course you can always use a slightly different symbol without the whole label pathway. But just to let you know of this potential issue.

**Comment by jonbakerfish** [comment]
Which value of the `batch_size` for optimizer's [`rescale_grad=(1.0/batch_size)`](https://github.com/dmlc/mxnet/blob/master/python/mxnet/model.py#L762) should be set in this case? Is it the batch_size of label?

**Comment by coolbrain** [comment]
What should the rescale_grad set in this situation?
